exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/Otel2PrometheusConverter.java

```diff
@@ -400,29 +402,44 @@ private Exemplars convertDoubleExemplars(List<DoubleExemplarData> exemplars) {
     return Exemplars.of(result);
   }
 
+  @Nullable
   private Exemplar convertExemplar(double value, ExemplarData exemplar) {
     SpanContext spanContext = exemplar.getSpanContext();
+    Labels labels = Labels.EMPTY;
     if (spanContext.isValid()) {
-      return new Exemplar(
-          value,
+      labels =
           convertAttributes(
               null, // resource attributes are only copied for point's attributes
               null, // scope attributes are only needed for point's attributes
               exemplar.getFilteredAttributes(),
               "trace_id",
               spanContext.getTraceId(),
               "span_id",
-              spanContext.getSpanId()),
-          exemplar.getEpochNanos() / NANOS_PER_MILLISECOND);
+              spanContext.getSpanId());
     } else {
-      return new Exemplar(
-          value,
-          convertAttributes(
-              null, // resource attributes are only copied for point's attributes
-              null, // scope attributes are only needed for point's attributes
-              exemplar.getFilteredAttributes()),
-          exemplar.getEpochNanos() / NANOS_PER_MILLISECOND);
+      labels = convertAttributes(null, null, exemplar.getFilteredAttributes());
+    }
+    int codePoints = getCodePoints(labels);
+    if (codePoints > EXEMPLAR_MAX_CODE_POINTS) {
+      THROTTLING_LOGGER.log(
+          Level.WARNING,
+          "exemplar labels have "
+              + codePoints
+              + " unicode code points, exceeding the limit of "
+              + EXEMPLAR_MAX_CODE_POINTS);
+      return null;
+    }
+    return new Exemplar(value, labels, exemplar.getEpochNanos() / NANOS_PER_MILLISECOND);
+  }
+
+  private static int getCodePoints(Labels labels) {
+    int codePoints = 0;
+    for (Label l : labels) {
```

Comment:
strongly suggest you rename l -> label. Otherwise it looks like the number '1'

Reply:
@jkwatson i don't think it's a big deal as the loop is too small, but i suppose it does help in quicker reading. I addressed it in latest commit.